bitcoin/src/bip32.rs

```diff
@@ -192,7 +192,9 @@ impl FromStr for ChildNumber {
     fn from_str(inp: &str) -> Result<ChildNumber, Error> {
         let is_hardened = inp.chars().last().map_or(false, |l| l == '\'' || l == 'h');
```

Comment:
Not relevant to the PR but this is linear where it could've been constant-time and written without indexing operations:


```rust
let trimmed = inp.trim_end_matches(&['\'', 'h'] as &[_]);`
// the second check prevents things like 42''''
if inp.is_empty() || trimmed.len() < inp.len() - 1 {
    Err(Error::InvalidChildNumberFormat)
} else {
    let idx = trimmed.parse().map_err(|_| Error::InvalidChildNumberFormat)?;
    if trimmed.len() < inp.len() {
        Ok(ChildNumber::from_hardened_idx(idx))
    } else {
        Ok(ChildNumber::from_normal_idx(idx))
    }
}
```